vql/linux/cronsnoop/plugin.go

```diff
@@ -0,0 +1,88 @@
+//go:build linux
+// +build linux
+
+package linux
+
+import (
+	"context"
+
+	"github.com/Velocidex/ordereddict"
+	"www.velocidex.com/golang/velociraptor/acls"
+	vql_subsystem "www.velocidex.com/golang/velociraptor/vql"
+	"www.velocidex.com/golang/vfilter"
+	"www.velocidex.com/golang/vfilter/arg_parser"
+)
+
+type CronsnoopArgs struct {
+	SpoolPath string `vfilter:"required,field=spool_dir,doc=Spool directory where user cron files are located"`
+	SystemPaths []string `vfilter:"required,field=system_dirs,doc=Directories to be watched where system cron files are located"`
+}
+
+type CronsnoopPlugin struct{}
+
+func (self CronsnoopPlugin) Info(scope vfilter.Scope, type_map *vfilter.TypeMap) *vfilter.PluginInfo {
+	return &vfilter.PluginInfo{
+		Name: "cronsnoop",
+		Doc:  "Snoops changes to cron files",
+	}
+}
+
+func (self CronsnoopPlugin) Call(
+	ctx context.Context, scope vfilter.Scope,
+	args *ordereddict.Dict) <-chan vfilter.Row {
+
+	output_chan := make(chan vfilter.Row)
+	arg := &CronsnoopArgs{}
+    err := arg_parser.ExtractArgsWithContext(ctx, scope, args, arg)
+
+	if err != nil {
+		scope.Log("cronsnoop: %s", err.Error())
+	}
+
+
+	go func() {
+		defer close(output_chan)
+
+		err := vql_subsystem.CheckAccess(scope, acls.MACHINE_STATE)
+		if err != nil {
+			scope.Log("cronsnoop: %s", err)
+			return
+		}
+
+		// Make the chan large enough so that cron snooper doesn't block, waiting 
+		// for the channel too be consumed
+		eventChan := make(chan CronEvent, 1000)
+		snooper, err := NewCronSnooperWithChan(arg.SpoolPath, arg.SystemPaths, eventChan)
```

Comment:
Oh the joys of random packages on the internet... I pulled in the lfreequeue and it turned out its watchIterator actually cannot be created, since the type is unexported (watchIterator vs WatchIterator, I guess the original author never tested this). Anyway, fixed that and indeed the lfreequeue can be used however, it leads to a slight inconvenience that the caller should be doing type assertions to get a CronEvent vs the generic interface {} with which lfreequeue has been defined. Of course this could be fixed by running a go routine that does the actual dequeue + type assertion and pumps stuff to properly typed channel and in turn this channel is the one returned to the user but at this point I'm starting to debate the utility of all this. One of the tests adjusted leads to the following: 

```
    // returns the WatchIterator's Iter()'s channel
    c := s.GetEventChan()                                                       
                                                                                                                                           
    v := <-c                       // reads a  generic interface {} typec                                         
    v1 := v.(CronEvent)     // do the typeassertion (casting)e                                                   
                                                                                
    assertCronEvent(v1, cmd[0], fName, fName, Added, t)                         
    v = <-c                                                                     
    v1 = v.(CronEvent)                                                          
    assertCronEvent(v1, cmd[1], fName, fName, Added, t)       
```